### PR TITLE
qcontainer:modify to choose right scsi_hba for scsi device on s390x platform

### PR DESCRIPTION
**Pull request by maxujun**
BZ:1486598
qcontainer:modify to choose right scsi_hba for scsi device on s390x platform
Signed-off-by: Xujun Ma <xuma@redhat.com>

### CODE REVIEW

File: virttest/qemu_devices/qcontainer.py [hunk 1]
```diff
@@ -1569,12 +1569,13 @@ def images_define_by_params(self, name, image_params, media=None,
...
-            if "s390" in machine_type:      # s390
+        machine_type = image_params.get("machine_type")
+        if "s390" in machine_type:      # s390
+            scsi_hba = "virtio-scsi-ccw"
+            if drive_format == "virtio":
                 drive_format = "virtio-blk-ccw"
-            elif "mmio" in machine_type:    # mmio-based machine
-                drive_format = "virtio-blk-device"
+        elif "mmio" in machine_type:    # mmio-based machine
+            drive_format = "virtio-blk-device"
```
Comment (yanan-fu) [inline virttest/qemu_devices/qcontainer.py]: Before, for `mmio`, it  only use `drive_format = "virtio-blk-device"` if `drive_format == "virtio"` .  and the new change remove it, this is not a correct logic.

File: virttest/qemu_devices/qcontainer.py [hunk 2]
```diff
@@ -1661,23 +1662,22 @@ def cdroms_define_by_params(self, name, image_params, media=None,
         image_params['image_raw_device'] = 'yes'
         cd_format = image_params.get('cd_format')
         scsi_hba = image_params.get("scsi_hba")
-        if cd_format is None or cd_format is 'ide':
```
Comment (yanan-fu) [inline virttest/qemu_devices/qcontainer.py]: Why did you remove `if cd_format is None or cd_format is 'ide'` ?   The following setting is based on cd_format = ide i think.

File: virttest/qemu_devices/qcontainer.py (hunk 1 above)
Comment (maxujun) [reply virttest/qemu_devices/qcontainer.py]: ok，keep the logic,and add right scsi_hba to another line.thanks for your comment.

File: virttest/qemu_devices/qcontainer.py (hunk 2 above)
Comment (maxujun) [reply virttest/qemu_devices/qcontainer.py]: same as above.

**Comment by yanan-fu** [comment]
LGTM
Could you add the test result when run with `only virtio_scsi`,  as it use `scsi_hba == "virtio-scsi-pci"` can cover the new change.

**Comment by maxujun** [comment]
@yanan-fu The vm command line created by avocado is：
QMP-catch_monitor',t'SER-serial0',t'isa-log',virtio_scsi_ccw0,drive_image1,image1,idAFPkf5,t'NET-bridge',t'mem',t'smp',t'cpu',t'display',t'rtc',t'bootmenu',t'kvm']
2017-09-12 22:22:42,969 qemu_vm          L2717 DEBUG| Buses of avocado-vt-vm1
  drive_image1(QDrive): {drive:image1}
  virtio_scsi_ccw0.0(virtio-scsi-ccw): {0-0:image1}
  virtio-blk-ccw(virtio-bus): {0:idAFPkf5}
2017-09-12 22:22:42,970 qemu_vm          L2781 INFO | Running qemu command (reformatted):
MALLOC_PERTURB_=1  /usr/libexec/qemu-kvm \
    -S  \
    -name 'avocado-vt-vm1'  \
    -sandbox off  \
    -machine s390-ccw-virtio  \
    -nodefaults  \
    -vga none  \
    -chardev socket,id=qmp_id_qmpmonitor1,path=/var/tmp/avocado_sEA050/monitor-qmpmonitor1-20170912-222241-diIIzEDt,server,nowait \
    -mon chardev=qmp_id_qmpmonitor1,mode=control  \
    -chardev socket,id=qmp_id_catch_monitor,path=/var/tmp/avocado_sEA050/monitor-catch_monitor-20170912-222241-diIIzEDt,server,nowait \
    -mon chardev=qmp_id_catch_monitor,mode=control  \
    -chardev socket,id=serial_id_serial0,path=/var/tmp/avocado_sEA050/serial-serial0-20170912-222241-diIIzEDt,server,nowait \
    -device sclpconsole,chardev=serial_id_serial0 \
    -device virtio-scsi-ccw,id=virtio_scsi_ccw0 \
    -drive id=drive_image1,if=none,snapshot=off,aio=native,cache=none,format=qcow2,file=/root/staf-kvm-devel/vt_test_images/ALT-Server-7.4-s390x-virtio-scsi.qcow2 \
    -device scsi-hd,id=image1,drive=drive_image1 \
    -device virtio-net-ccw,mac=9a:53:54:55:56:57,id=idAFPkf5,netdev=idm3EwjZ  \
    -netdev tap,id=idm3EwjZ,vhost=on,vhostfd=20,fd=19 \
    -m 1515  \
    -smp 2,cores=1,threads=1,sockets=2  \
    -cpu 'host'  \
    -vnc :0  \
    -rtc base=utc,clock=host,driftfix=slew \
    -boot strict=on \
    -enable-kvm

**Comment by yanan-fu** [comment]
@maxujun i think the test result only cover disk, no cdrom.  The change in `cdroms_define_by_params` was not covered.

**Comment by maxujun** [comment]
@yanan-fu 
MALLOC_PERTURB_=1  /usr/libexec/qemu-kvm \
    -S  \
    -name 'avocado-vt-vm1'  \
    -sandbox off  \
    -machine s390-ccw-virtio  \
    -nodefaults  \
    -vga none  \
    -chardev socket,id=qmp_id_qmpmonitor1,path=/var/tmp/avocado_sEA050/monitor-qmpmonitor1-20170912-210202-5n8NlJea,server,nowait \
    -mon chardev=qmp_id_qmpmonitor1,mode=control  \
    -chardev socket,id=qmp_id_catch_monitor,path=/var/tmp/avocado_sEA050/monitor-catch_monitor-20170912-210202-5n8NlJea,server,nowait \
    -mon chardev=qmp_id_catch_monitor,mode=control  \
    -chardev socket,id=serial_id_serial0,path=/var/tmp/avocado_sEA050/serial-serial0-20170912-210202-5n8NlJea,server,nowait \
    -device sclpconsole,chardev=serial_id_serial0 \
    -device virtio-scsi-ccw,id=virtio_scsi_ccw0 \
    -drive id=drive_image1,if=none,snapshot=off,aio=threads,cache=unsafe,format=qcow2,file=/root/staf-kvm-devel/vt_test_images/ALT-Server-7.4-s390x-virtio-scsi.qcow2 \
    -device scsi-hd,id=image1,drive=drive_image1 \
    -device virtio-net-ccw,mac=9a:53:54:55:56:57,id=iduBPC7k,netdev=id3sz7eS  \
    -netdev tap,id=id3sz7eS,vhost=on,vhostfd=21,fd=20 \
    -m 1515  \
    -smp 2,cores=1,threads=1,sockets=2  \
    -cpu 'host' \
    -drive id=drive_cd1,if=none,snapshot=off,aio=threads,cache=unsafe,media=cdrom,file=/var/lib/avocado/data/avocado-vt/isos/linux/RHEL-ALT-7.4-20170808.0-Server-s390x-dvd1.iso \
    -device scsi-cd,id=cd1,drive=drive_cd1 \
    -drive id=drive_unattended,if=none,snapshot=off,aio=threads,cache=unsafe,media=cdrom,file=/root/staf-kvm-devel/vt_test_images/rhel-alt-74-s390x/ks.iso \
    -device scsi-cd,id=unattended,drive=drive_unattended  \
    -kernel '/root/staf-kvm-devel/vt_test_images/rhel-alt-74-s390x/kernel.img'  \
    -append 'ksdevice=link inst.repo=cdrom:/dev/sr0 inst.ks=cdrom:/dev/sr1:/ks.cfg nicdelay=60 biosdevname=0 net.ifnames=0 console=ttysclp0'  \
    -initrd '/root/staf-kvm-devel/vt_test_images/rhel-alt-74-s390x/initrd.img'  \
    -vnc :0  \
    -rtc base=utc,clock=host,driftfix=slew \
    -boot strict=on  \
    -no-shutdown \
    -enable-kvm

**Comment by yanan-fu** [comment]
LGTM
Acked-by: Yanan Fu <yfu@redhat.com>
